receiver/opencensusreceiver/opencensus_test.go

```diff
@@ -179,6 +179,36 @@ func TestGrpcGatewayCors_endToEnd(t *testing.T) {
 	verifyCorsResp(t, url, "disallowed-origin.com", 200, false)
 }
 
+func TestMetricsGrpcGatewayCors_endToEnd(t *testing.T) {
+	addr := ":35991"
+	corsOrigins := []string{"allowed-*.com"}
+
+	sink := new(exportertest.SinkMetricsExporter)
+	ocr, err := New(addr, nil, sink, WithCorsOrigins(corsOrigins))
+	if err != nil {
+		t.Fatalf("Failed to create metrics receiver: %v", err)
+	}
+	defer ocr.StopMetricsReception()
+
+	mh := receivertest.NewMockHost()
+	go func() {
```

Comment:
nit: this doesn't need to be in a goroutine

Reply:
IIRC `StartMetricsReception` is blocking so it needs to be in a goroutine.

Reply:
no, they are not blocking. They are expected to let the service running in the background, we should document this fact on the interface (at least).

Reply:
I see, updated the tests here.